fe/fe-core/src/main/java/com/starrocks/sql/spm/SQLPlanGlobalStorage.java

```diff
@@ -235,6 +236,39 @@ public void replayBaselinePlan(BaselinePlan.Info info, boolean isCreate) {
         }
     }
 
+    public void replayUpdateBaselinePlan(BaselinePlan.Info info, boolean isEnable) {
+        // invalidate cache
+        cache.invalidateAll(info.getReplayIds());
+    }
+
+    @Override
+    public void controlBaselinePlan(boolean isEnable, List<Long> baseLineIds) {
+        if (!StatisticUtils.checkStatisticTables(List.of(StatsConstants.SPM_BASELINE_TABLE_NAME))) {
+            return;
+        }
+        List<Long> ids = allBaselineIds.stream()
+                .filter(b -> baseLineIds.contains(b.id))
+                .map(BaselineId::id)
+                .toList();
+        if (ids.isEmpty()) {
+            return;
+        }
+
+        String sql = UPDATE_SQL + isEnable + " WHERE id IN (" +
```

Comment:
how do we use `enable` flag?

Reply:
in `SPMPlanner.java`, rewrite sql by enable baseline only